cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -198,16 +142,14 @@ impl<State> CoreEnvironment<State> {
     }
 
     /// Lock the environment if it isn't already locked.
-    /// Error if there is a v0 manifest that needs to be locked, because
-    /// locking with pkgdb is no longer supported
     ///
     /// This might be a slight optimization as compared to calling [Self::lock],
     /// but [Self::lock] skips re-locking already locked packages,
     /// so it probably doesn't make much of a difference.
     /// The real point of this method is letting us skip locking for an already
     /// locked pkgdb manifest,
     /// since pkgdb manifests can no longer be locked.
```

Comment:
Related to https://github.com/flox/flox/issues/2202#issuecomment-2397648709 we do have [low-level test coverage](https://github.com/flox/flox/blob/53962a0511edf9c4ae33597f103f56e78342b32a/cli/flox-rust-sdk/src/models/environment/core_environment.rs#L2050-L2068) and I think the previous refactoring we did for `ensure_locked`, `lock`, and `build` (which takes lock content without any implicit locking) is sufficient to prevent us regressing.

```suggestion
    /// This is a slight optimization as compared to calling [Self::lock], even
    /// though [Self::lock] skips re-locking already locked packages, but it
    /// also makes it clearer which code paths may require re-locking.
```